src/App.jsx

```diff
@@ -1,54 +1,59 @@
+import Login from "./pages/login";
+import Register from "./pages/register";
+import Loading from "./pages/loading";
+import Verification from "./pages/verification";
+import Profile from "./pages/profile";
+import { AuthProvider, ProtectedRoute } from "./context/auth";
+import { ModalProvider } from "./context/modal";
+import { CurrentUserProvider } from "./context/currentUser";
+import Welcome from "./pages/welcome";
+import AllSearchResults from "./pages/allSearchResults";
 
 const App = () => {
-    return (
-        <>
-            <AuthProvider>
-                <CurrentUserProvider>
-                    <ModalProvider>
-                        <Routes>
-                            <Route path="login" element={<Login />} />
-                            <Route path="register" element={<Register />} />
-                            <Route path="loading" element={<Loading />} />
-                            <Route
-                                path="verification"
-                                element={<Verification />}
-                            />
+  return (
+    <>
+      <AuthProvider>
+        <CurrentUserProvider>
+          <ModalProvider>
+            <Routes>
+              <Route path="login" element={<Login />} />
+              <Route path="register" element={<Register />} />
+              <Route path="loading" element={<Loading />} />
+              <Route path="verification" element={<Verification />} />
 
-                            <Route
-                                index
-                                element={
-                                    <ProtectedRoute>
-                                        <Dashboard />
-                                    </ProtectedRoute>
-                                }
-                            />
-                            <Route
-                                path="welcome"
-                                element={
-                                    <ProtectedRoute disabledNav={true}>
-                                        <Welcome />
-                                    </ProtectedRoute>
-                                }
-                            />
-                            <Route path="/search-results" element={<AllSearchResults />} />
-                        </Routes>
-                    </ModalProvider>
-                </CurrentUserProvider>
-            </AuthProvider>
-        </>
-    )
-}
+              <Route
+                index
+                element={
+                  <ProtectedRoute>
+                    <Dashboard />
+                  </ProtectedRoute>
+                }
+              />
+              <Route
+                path="welcome"
+                element={
+                  <ProtectedRoute disabledNav={true}>
+                    <Welcome />
+                  </ProtectedRoute>
+                }
+              />
+              <Route path="/search-results" element={<AllSearchResults />} />
```

Comment:
I know this isn't yours, but shouldn't this be behind a protected route? If you think it should too, can you add it as a ticket to be worked on please?

Reply:
Sorry for popping up here. I did this. Why should it be in a protected route?

Reply:
Should you be able to see search results, or the search results page, without being logged in?

Reply:
Hmmm. You can't see results, but you can get to the page. Should I fix it or or Will will?

Reply:
I leave that up to your discretion 😄

Reply:
This is a ticket now, here: https://github.com/boolean-uk/team-dev-client-c12/issues/142